packages/compass-sidebar/src/components/multiple-connections/connections-navigation.tsx

```diff
@@ -0,0 +1,466 @@
+  type Database,
+} from '../../modules/databases';
+import { useFilteredConnections } from '../use-filtered-connections';
+import NavigationItemsFilter from '../navigation-items-filter';
+
+const connectionsContainerStyles = css({
+  height: '100%',
+  paddingLeft: spacing[400],
+  paddingRight: spacing[400],
+  paddingBottom: spacing[400],
+  display: 'flex',
+  flexDirection: 'column',
+  gap: spacing[200],
+});
+
+const connectionListHeaderStyles = css({
+  display: 'flex',
+  justifyContent: 'space-between',
+  alignItems: 'center',
+});
+
+const connectionListHeaderTitleStyles = css({
+  marginTop: 0,
+  marginBottom: 0,
+  textTransform: 'uppercase',
+  fontSize: '12px',
+  lineHeight: `${spacing[800]}px`,
+});
+
+const connectionCountStyles = css({
+  fontWeight: 'normal',
+  marginLeft: spacing[100],
+});
+
+function findCollection(ns: string, databases: Database[]) {
+  const { database, collection } = toNS(ns);
+
+  return (
+    databases
+      .find((db) => db._id === database)
+      ?.collections.find((coll) => coll.name === collection) ?? null
+  );
+}
+
+type ConnectionsNavigationComponentProps = {
+  connectionsWithStatus: ReturnType<typeof useConnectionsWithStatus>;
+  activeWorkspace: WorkspaceTab | null;
+  filterRegex: RegExp | null;
+  onFilterChange(regex: RegExp | null): void;
+  onConnect(info: ConnectionInfo): void;
+  onEditConnection(info: ConnectionInfo): void;
+  onRemoveConnection(info: ConnectionInfo): void;
+  onDuplicateConnection(info: ConnectionInfo): void;
+  onCopyConnectionString(info: ConnectionInfo): void;
+  onToggleFavoriteConnectionInfo(info: ConnectionInfo): void;
+
+  onOpenConnectionInfo(id: string): void;
+  onDisconnect(id: string): void;
+};
+
+type MapStateProps = {
+  instances: RootState['instance'];
+  databases: RootState['databases'];
+  isPerformanceTabSupported: RootState['isPerformanceTabSupported'];
+};
+
+type MapDispatchProps = {
+  fetchAllCollections(): void;
+  onDatabaseExpand(connectionId: string, dbId: string): void;
+  onNamespaceAction(
+    connectionId: string,
+    namespace: string,
+    action: Actions
+  ): void;
+};
+
+type ConnectionsNavigationProps = ConnectionsNavigationComponentProps &
+  MapStateProps &
+  MapDispatchProps;
+
+const ConnectionsNavigation: React.FC<ConnectionsNavigationProps> = ({
+  connectionsWithStatus,
+  activeWorkspace,
+  filterRegex,
+  instances,
+  databases,
+  isPerformanceTabSupported,
+  onFilterChange,
+  onConnect,
+  onEditConnection,
+  onRemoveConnection,
+  onDuplicateConnection,
+  onCopyConnectionString,
+  onToggleFavoriteConnectionInfo,
+
+  onOpenConnectionInfo,
+  onDisconnect,
+  onDatabaseExpand,
+  fetchAllCollections,
+  onNamespaceAction: _onNamespaceAction,
+}) => {
+  const {
+    openShellWorkspace,
+    openPerformanceWorkspace,
+    openDatabasesWorkspace,
+    openCollectionsWorkspace,
+    openCollectionWorkspace,
+    openEditViewWorkspace,
+  } = useOpenWorkspace();
+  const connections = useMemo(() => {
```

Comment:
So about this memo - it could have been in the mapState but I opted for memo in the hopes of avoiding re-calculation on every irrelevant change in the sidebar store.
I understand that it means I need to pass down the entire database, instance and isPerformanceTabSupported slice to this component but the other option was to refactor the store a little and perform this calculation within the store.

But to be honest I also did not spend too much time on this so please feel free to recommend something for this. I am open to suggestions.